include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -347,24 +347,80 @@ MultigridPreconditionerBase<dim, Number>::initialize_mapping()
   {
     if(data.use_global_coarsening) // global coarsening
     {
-      unsigned int const n_h_levels = coarse_grid_triangulations.size();
+      FESystem<dim>                          fe(FE_Q<dim>(mapping_q_cache->get_degree()), dim);
+      unsigned int const                     n_h_levels = coarse_grid_triangulations.size();
+      std::vector<DoFHandler<dim>>           coarse_grid_dof_handlers(n_h_levels);
+      std::vector<AffineConstraints<Number>> coarse_grid_constraints(n_h_levels);
+      for(unsigned int i = 0; i < n_h_levels; ++i)
+      {
+        coarse_grid_dof_handlers[i].reinit(*coarse_grid_triangulations[i]);
+        coarse_grid_dof_handlers[i].distribute_dofs(fe);
+        // constraints are irrelevant for interpolation
+        coarse_grid_constraints[i].close();
```

Comment:
Can't we have hanging nodes on coarse levels that might lead to gaps in the coarse geometry if we interpolate plain down? Or do we fix up hanging nodes in a similar code as here: https://gitlab.lrz.de/exadg/exadg-applications/-/blob/master/applications/lung/include/grid/mapping/initialize_mapping.h#L56-58
(but I did not find any).

Not that I expect a big influence on the multigrid convergence, as this is only an auxiliary mesh in some preconditioner, so there is absolutely no physical change in the solution.